NativeYoutube/Views/SharedViews/VideoRowView.swift

```diff
@@ -16,25 +15,38 @@ struct VideoRowView: View {
     var body: some View {
         Group {
             ZStack {
-                WebImage(url: video.thumbnail)
-                    .resizable()
-                    .overlay {
-                        Rectangle()
-                            .fill(hovered ? .ultraThinMaterial : .ultraThickMaterial)
-                    }
+                AsyncImage(url: video.thumbnail, content: { image in
+                    image
+                        .resizable()
+                        .overlay {
+                            RoundedRectangle(cornerRadius: 5)
+                                .fill(hovered ? .ultraThinMaterial : .ultraThickMaterial)
+                        }
+                }, placeholder: {
+                    Spacer()
+                        .overlay {
+                            RoundedRectangle(cornerRadius: 5)
+                                .fill(hovered ? .ultraThinMaterial : .ultraThickMaterial)
+                        }
+                })
 
-                HStack {
-                    if !hovered {
-                        WebImage(url: video.thumbnail)
+                HStack() {
+                    AsyncImage(url: video.thumbnail, content: { image in
```

Comment:
AsyncImage doesn't support caching.

There's a reason I've opt-ed to using SDWEBImage SwiftUI!

Reply:
Yup, as i explained in the pr, i figured out after modifying that ofc there is a reason for switching xD
Should be added back with the latest push commit !